AppCenterData/AppCenterData/Internal/Client/MSDataOperationProxy.m

```diff
@@ -207,12 +207,19 @@ - (void)performListOnDocumentType:(Class)documentType
                                                                        documentType:documentType
                                                                         baseOptions:baseOptions];
 
+      if ([self.reachability currentReachabilityStatus] == ReachableViaWiFi && [[cachedDocumentsList currentPage] items].count == 0) {
```

Comment:
It's better to check that the status is not unrechable. This test does not include cellular reachability I guess.

Reply:
sounds like a bug to me